### PR TITLE
reduce bytearray copy to minimal optimize VSizeIndexedWriter

### PR DESCRIPTION
**Pull request by binlijin**

### CODE REVIEW

File: processing/src/main/java/io/druid/segment/data/VSizeIndexedInts.java [hunk 1]
```diff
@@ -50,6 +50,31 @@ public static VSizeIndexedInts empty()
     return fromList(Lists.<Integer>newArrayList(), 0);
   }
 
+  /**
+   * provide for performance reason.
+   */
+  public static byte[] getBytesNoPaddingfromList(List<Integer> list, int maxValue)
+  {
+    int numBytes = getNumBytesForMax(maxValue);
```
Comment (xvrl) [inline processing/src/main/java/io/druid/segment/data/VSizeIndexedInts.java]: can we refactor to reduce copy-paste with `fromList`?

**Comment by xvrl** [comment]
@binlijin any numbers on how much this improves performance?

**Comment by binlijin** [comment]
@xvrl , the performance numbers is：
Before：
round1
2015-12-10T15:49:19,781 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base8884394285534279712flush/merged/v8-tmp] walked 500,000/500,000 rows in 16,840 millis.
2015-12-10T15:49:40,627 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base8884394285534279712flush/merged/v8-tmp] completed walk through of 916,822 rows in **37,767** millis.
round2
2015-12-10T15:50:59,429 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base774856129942668621flush/merged/v8-tmp] walked 500,000/500,000 rows in 16,473 millis.
2015-12-10T15:51:19,962 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base774856129942668621flush/merged/v8-tmp] completed walk through of 916,822 rows in **37,065** millis.

After：
round1
2015-12-10T15:52:41,049 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base4028347591001450805flush/merged/v8-tmp] walked 500,000/500,000 rows in 13,857 millis.
2015-12-10T15:52:58,643 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base4028347591001450805flush/merged/v8-tmp] completed walk through of 916,822 rows in **31,505** millis.
round2
2015-12-10T15:54:14,273 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base4543915257477602206flush/merged/v8-tmp] walked 500,000/500,000 rows in 14,390 millis.
2015-12-10T15:54:31,960 INFO [main] io.druid.segment.IndexMerger - outDir[/var/folders/z7/g5zy3kfj7t54y1f074hsqw7h0000gn/T/base4543915257477602206flush/merged/v8-tmp] completed walk through of 916,822 rows in **32,169** millis.

**Comment by nishantmonu51** [comment]
:+1:

**Pull request merged**